cloudinit/sources/helpers/vultr.py

```diff
@@ -21,8 +21,8 @@ def get_metadata(url, timeout, retries, sec_between, agent):
 
     # Seek iface with DHCP
     for iface in net.get_interfaces():
-        # Skip dummy interfaces
-        if "dummy" in iface[0]:
+        # Skip dummy, lo interfaces
+        if iface[0] in ['lo', 'dummy']:
```

Comment:
This check would now fail to filter out numbered dummy interfaces, e.g. "dummy0".

FWIW I've been considering taking the filtering done for net.find_fallback_nic() and providing a method so we can query the set of viable interfaces without the data sources having to add similar filtering logic for themselves if fallback_interface is not sufficient.

Reply:
Ah indeed it would. Not sure why I didn't see that... Anyways its fixed, and it looks like find_candidate_nics is exactly what I need. Once that's merged in I will immediately refactor this to use that instead. Probably when I go through an simplify the generation of net metadata. I am looking at making a new endpoint on our end that feeds cloud-init format configs directly instead of doing what we are doing here to eliminate some of these inconsistencies we are seeing due to our metadata bumping heads with the cloud-init format here and there.